### PR TITLE
Add FOREIGN_GAS_PRICE_ORACLE_URL to env file and checking to qs

### PR DESCRIPTION
**Pull request by Andrew-Pohl**

### CODE REVIEW

**Review by leonprou — APPROVED** [review]

**Review by leonprou — CHANGES_REQUESTED** [review]
Still not sure if it's cool to require from all the validators to register on the `defipulse`

**Comment by Andrew-Pohl** [comment]
> Still not sure if it's cool to require from all the validators to register on the `defipulse`

Hmmm alternatively we could grab the gas price from the Eth nodes (but this isn't as accurate). Or we could just update the fallback oracle to use a new API key. But then we are back to a single point of failure

**Comment by leonprou** [comment]
@Andrew-Pohl I suggest that we try to call the provided endpoint by `FOREIGN_GAS_PRICE_ORACLE_URL` in the quickstart and if it's good we pass that check.
ideally need to check that the response is status 200 and you get a json with "fast" field in it, like:
```
{"fast": 410.0, "fastest": 450.0, ...}
```
you already check that status code there

**Comment by Andrew-Pohl** [comment]
> @Andrew-Pohl I suggest that we try to call the provided endpoint by `FOREIGN_GAS_PRICE_ORACLE_URL` in the quickstart and if it's good we pass that check.
> ideally need to check that the response is status 200 and you get a json with "fast" field in it, like:
> 
> ```
> {"fast": 410.0, "fastest": 450.0, ...}
> ```
> 
> you already check that status code there. what you think?

Ok I see what you mean, got a little confused there so make the FOREIGN_GAS_PRICE_ORACLE_URL mandatory but allow any provider i.e. remove https://github.com/fuseio/fuse-network/blob/9d434559ca337324450e61b45410d8a248d53953/scripts/quickstart.sh#L139 this check. Only issue I see is that other providers give different units... we currently scale ethGasAPI by dividing it by 10 (they give the value as gwei * 10). Not sure how we'd know what to scale it by correctly.

**Comment by leonprou** [comment]
@Andrew-Pohl yep I see it's problematic.. But I also afraid  that validators will run the quickstart and got stuck on this param.
I suggest let's make it a warning if it doesn't exists, but not an exit error

**Comment by Andrew-Pohl** [comment]
> @Andrew-Pohl yep I see it's problematic.. But I also afraid that validators will run the quickstart and got stuck on this param.
> I suggest let's make it a warning if it doesn't exists, but not an exit error

@leonprou  Understood! I will make it still only accept ethgas station API. If another is given I will default back to the old ethgas api (to avoid the scenario where the units are different and the gas price gets set to something insanely high!)

Will also buffer all the warnings and display them at the end to avoid them getting lost in the midst of everything 👍

**Review by leonprou — APPROVED** [review]
@Andrew-Pohl Thanks for continue tweaking it 😅

File: scripts/quickstart.sh [hunk 1]
```diff
@@ -130,6 +133,33 @@ function getAndUpdateBlockNumbers {
...
+        resetOracle=true
+      fi
+    else
+      WARNINGS+=("FOREIGN_GAS_PRICE_ORACLE_URL Does not match ethgasstation endpoint see https://data.defipulse.com/ for more details, using the default oracle, recommend to create your own!")
+      resetOracle=true
+    fi
+
+    if [ "$resetOracle" = true ] ; then
+      echo "Reset FOREIGN_GAS_PRICE_ORACLE_URL back to default"
+      sed -i "s/^FOREIGN_GAS_PRICE_ORACLE_URL.*/FOREIGN_GAS_PRICE_ORACLE_URL=$DEFAULT_GAS_ORACLE/" "$ENV_FILE"
```
Comment (leonprou) [inline scripts/quickstart.sh:158]: @Andrew-Pohl I had a late idea on that, if you just put an empty string in  `FOREIGN_GAS_PRICE_ORACLE_URL`, it'll use the default value in the container, right? So we don't need to keep the oracle url in multiple places

Comment (Andrew-Pohl) [reply scripts/quickstart.sh:158]: That's a very good point!, If it's incorrect we can just delete it from this env file and let the default do its job! 👍

**Pull request merged**